nixos/modules/services/desktop-managers/cosmic.nix

```diff
@@ -0,0 +1,166 @@
+{ config, lib, pkgs, utils, ... }:
+
+let
+  cfg = config.services.desktopManager.cosmic;
+in
+{
+  meta.maintainers = with lib.maintainers; [
+    thefossguy
+  ];
+
+  options = {
+    services.desktopManager.cosmic = {
+      enable = lib.mkEnableOption "COSMIC desktop environment";
+
+      unstable.protocols.enable = lib.mkEnableOption "Enable the [currently unstable] cosmic-protocols" // {
+        default = false;
+      };
+
+      xwayland.enable = lib.mkEnableOption "Xwayland support for cosmic-comp" // {
+        default = true;
+      };
+    };
+
+    environment.cosmic.excludePackages = lib.mkOption {
+      description = "List of COSMIC packages to exclude from the default environment";
+      type = lib.types.listOf lib.types.package;
+      default = [ ];
+      example = lib.literalExpression "[ pkgs.cosmic-player ]";
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    # Environment packages
+    environment.pathsToLink = [
+      "/share/backgrounds"
+      "/share/cosmic"
+    ];
+    environment.systemPackages = utils.removePackagesByName (
+      with pkgs;
+      [
+        adwaita-icon-theme
+        alsa-utils
+        cosmic-applets
+        cosmic-applibrary
+        cosmic-bg
+        cosmic-comp
+        cosmic-edit
+        cosmic-files
+        cosmic-greeter
+        cosmic-icons
+        cosmic-idle
+        cosmic-launcher
+        cosmic-notifications
+        cosmic-osd
+        cosmic-panel
+        cosmic-player
+        cosmic-randr
+        cosmic-reader
+        cosmic-screenshot
+        cosmic-session
+        cosmic-settings
+        cosmic-settings-daemon
+        cosmic-term
+        cosmic-wallpapers
+        cosmic-workspaces-epoch
+        hicolor-icon-theme
+        playerctl
+        pop-icon-theme
+        pop-launcher
+        xdg-user-dirs
+      ]
+      ++ lib.optionals cfg.unstable.protocols.enable [
+        cosmic-protocols
+      ]
+      ++ lib.optionals cfg.xwayland.enable [
+        xwayland
+      ]
+      ++ lib.optionals config.services.flatpak.enable [
+        cosmic-store
+      ]
+    ) config.environment.cosmic.excludePackages;
+
+    xdg.portal = {
+      enable = true;
+      extraPortals = lib.mkBefore [ pkgs.xdg-desktop-portal-cosmic ];
+      configPackages = lib.mkDefault ([ pkgs.xdg-desktop-portal-cosmic ]);
```

Comment:
What is mkDefault solving here? It should be matching the prio level of other DEs